### PR TITLE
Document acting and enumerable semigroups

### PR DESCRIPTION
**Pull request by james-d-mitchell**
This PR replaces #300 and depends on PR #304.

### CODE REVIEW

File: doc/fropin.xml [hunk 1]
```diff
@@ -28,3 +37,173 @@ false
...
+<#GAPDoc Label="AsListCanonical">
+  <ManSection>
+    <Attr Name="AsListCanonical" Arg="S"/> 
+    <Attr Name="EnumeratorCanonical" Arg="S"/>
+    <Attr Name="IteratorCanonical" Arg="S"/>
+    <Returns>A list, enumerator, or iterator.</Returns>
+    <Description>
+      When the argument <A>S</A> is a semigroup in the representation 
+      <Ref Filt="IsEnumerableSemigroupRep"/>, <C>AsListCanonical</C>
+      returns a list of elements of <A>S</A> in the order they are enumerated
```
Comment (wilfwilson) [inline doc/fropin.xml]: I recommend adding "the": `a list of the elements` - just to make it clear that it contains all elements.

File: doc/fropin.xml [hunk 2]
```diff
@@ -28,3 +37,173 @@ false
...
+  </ManSection>
+<#/GAPDoc>
+
+<#GAPDoc Label="Enumerate">
+  <ManSection>
+    <Oper Name="Enumerate" Arg="S[, limit]"/> 
+    <Returns>A semigroup (the argument).</Returns>
+    <Description>
+      If <A>S</A> is a semigroup with representation 
+      <Ref Filt="IsEnumerableSemigroupRep"/> and <A>limit</A> is in a positive
```
Comment (wilfwilson) [inline doc/fropin.xml]: "is in a positive" -> "is a positive"

File: doc/fropin.xml [hunk 3]
```diff
@@ -28,3 +37,173 @@ false
...
+      For reasons of performance, <A>S</A> is enumerated in batches according
+      to the option <C>batch_size</C>, which can be specified when <A>S</A> is
+      created; see Section <Ref Sect="Options when creating semigroups"/>.
+
+      <Example><![CDATA[
+gap> S := FullTransformationMonoid(7);
+<full transformation monoid of degree 7>
+gap> Enumerate(S, 1000);
+<full transformation monoid of degree 7>
+gap> Display(S); 0;
```
Comment (wilfwilson) [inline doc/fropin.xml]: Why is this `0;` here?

File: doc/z-chap06.xml [hunk 4]
```diff
@@ -7,50 +7,181 @@
...
+            \ldots, t, \infty\}</M> for some threshold <M>t</M> with
+          operations min and plus;
+        </Item>
+        <!-- <Item>
+          the <E>projective max-plus semiring</E> ... TODO
+        </Item> -->
+        <Item>
+          the semiring <M>\mathbb{N}_{t, p} = \{0, 1,
+            \ldots, t, t + 1, \ldots, t + p - 1\}</M> for some threshold
+          <M>t</M> and period <M>p</M> under addition and mutliplication
```
Comment (wilfwilson) [inline doc/z-chap06.xml]: "mutliplication" -> "multiplication"

**Comment by james-d-mitchell** [comment]
All done I think @wilfwilson

**Pull request merged**